calico/felix/test/test_config.py

```diff
@@ -0,0 +1,214 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 Metaswitch Networks
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+felix.test.test_config
+~~~~~~~~~~~
+
+Top level tests for Felix configuration.
+"""
+import logging
+import mock
+import socket
+import sys
+from calico.felix.config import Config, ConfigException
+
+if sys.version_info < (2, 7):
+    import unittest2 as unittest
+else:
+    import unittest
+
+import etcd
+real_client = etcd.client
+
+# Logger
+log = logging.getLogger(__name__)
+
+
+# Set of results from etcd.
+etcd_results = {}
+
+from collections import namedtuple
+EtcdChild = namedtuple('EtcdChild', ['key', 'value'])
+
+class StubEtcdResult(object):
+    def __init__(self, path):
+        self.children = []
+        self.path = path
+        etcd_results[path] = self
+
+    def add_child(self, key, value):
+        key = "%s/%s" % (self.path, key)
+        self.children.append(EtcdChild(key, value))
+
+class StubEtcd(object):
+    """
+    Trivial etcd stub.
+    """
+    def __init__(self, port=None):
+        self.port = port
+
+    def read(self, path):
+        return etcd_results[path]
+
+
+class TestConfig(unittest.TestCase):
+    def setUp(self):
+        # Stub out etcd.
+        etcd.Client = StubEtcd
+        etcd_results.clear()
+
+    def tearDown(self):
+        # Unstub etcd.
+        etcd.Client = real_client
+
+    def test_default_config(self):
+        """
+        Test various ways of defaulting config.
+        """
+        files = [ "felix_missing.cfg", # does not exist
+                  "felix_empty.cfg", # empty file
+                  "felix_empty_section.cfg", # file with empty section
+                  "felix_extra.cfg", # extra config is just logged
+                  ]
+
+        for file in files:
```

Comment:
I would replace this with `for filename in files` to avoid overriding the builtin file.

I don’t think it would cause any problems here, but it can’t hurt to be safe.